bftengine/src/bftengine/ClientsManager.cpp

```diff
@@ -176,31 +176,39 @@ std::unique_ptr<ClientReplyMsg> ClientsManager::allocateNewReplyMsgAndWriteToSto
                   << KVLOG(c.repliesInfo.size(), maxNumOfReqsPerClient_, clientId, requestSeqNum, replyLength));
   }
 
-  c.repliesInfo.insert_or_assign(requestSeqNum, getMonotonicTime());
   LOG_DEBUG(CL_MNGR, KVLOG(clientId, requestSeqNum));
-  auto r = std::make_unique<ClientReplyMsg>(myId_, requestSeqNum, reply, replyLength);
+  auto r = std::make_shared<ClientReplyMsg>(myId_, requestSeqNum, reply, replyLength);
+  c.repliesInfo.insert({requestSeqNum, r});
```

Comment:
Will not be re-written, if already exists.

Reply:
Changed to 'insert_or_assign' method.